winterfell/src/tests/logup_gkr_periodic.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use std::{marker::PhantomData, vec, vec::Vec};
+
+use air::{
+    Air, AirContext, Assertion, AuxRandElements, ConstraintCompositionCoefficients, FieldExtension,
+    LogUpGkrEvaluator, LogUpGkrOracle, ProofOptions, TraceInfo,
+};
+use crypto::MerkleTree;
+use math::StarkField;
+
+use super::super::*;
+use crate::{
+    crypto::{hashers::Blake3_256, DefaultRandomCoin},
+    math::{fields::f64::BaseElement, ExtensionOf, FieldElement},
+    matrix::ColMatrix,
+    DefaultConstraintEvaluator, DefaultTraceLde, Prover, StarkDomain, TracePolyTable,
+};
+
+#[test]
+fn test_logup_gkr_periodic() {
+    let aux_trace_width = 1;
+    let trace = LogUpGkrPeriodic::new(2_usize.pow(7), aux_trace_width);
+    let prover = LogUpGkrPeriodicProver::new(aux_trace_width);
+
+    let proof = prover.prove(trace).unwrap();
+
+    verify::<
+        LogUpGkrPeriodicAir,
+        Blake3_256<BaseElement>,
+        DefaultRandomCoin<Blake3_256<BaseElement>>,
+        MerkleTree<Blake3_256<BaseElement>>,
+    >(proof, (), &AcceptableOptions::MinConjecturedSecurity(0))
+    .unwrap()
+}
+
+// LogUpGkrPeriodic
+// =================================================================================================
+
+#[derive(Clone, Debug)]
+struct LogUpGkrPeriodic {
+    // dummy main trace
+    main_trace: ColMatrix<BaseElement>,
+    info: TraceInfo,
+}
+
+impl LogUpGkrPeriodic {
+    fn new(trace_len: usize, aux_segment_width: usize) -> Self {
+        assert!(trace_len < u32::MAX.try_into().unwrap());
+
+        let table: Vec<BaseElement> =
+            (0..trace_len).map(|idx| BaseElement::from(idx as u32)).collect();
+        let mut multiplicity: Vec<BaseElement> =
+            (0..trace_len).map(|_idx| BaseElement::ZERO).collect();
+        multiplicity.iter_mut().step_by(8).for_each(|m| *m = BaseElement::from(3_u32));
+
+        let mut values_0: Vec<BaseElement> = (0..trace_len).map(|_idx| BaseElement::ZERO).collect();
+
+        for i in 0..trace_len / 8 {
+            values_0[8 * i] = BaseElement::from(8 * i as u32);
+        }
+
+        let mut values_1: Vec<BaseElement> = (0..trace_len).map(|_idx| BaseElement::ZERO).collect();
+
+        for i in 0..trace_len / 8 {
+            values_1[8 * i] = BaseElement::from(8 * i as u32);
+        }
+
+        let mut values_2: Vec<BaseElement> = (0..trace_len).map(|_idx| BaseElement::ZERO).collect();
+
+        for i in 0..trace_len / 8 {
+            values_2[8 * i] = BaseElement::from(8 * i as u32);
+        }
+
+        let mut periodic: Vec<BaseElement> = (0..trace_len).map(|_idx| BaseElement::ZERO).collect();
+
+        for i in 0..trace_len / 8 {
+            periodic[8 * i] = BaseElement::ONE;
+        }
```

Comment:
`periodic` is never used. Was this intended?

Reply:
Yep, it was a leftover from some consistency checks I had before. Removed